Source/NSPersonNameComponentsFormatter.m

```diff
@@ -0,0 +1,236 @@
+/* Definition of class NSPersonNameComponentsFormatter
+   Copyright (C) 2019 Free Software Foundation, Inc.
+   
+   Implemented by: Gregory Casamento <greg.casamento@gmail.com>
+   Date: Sep 2019
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02111 USA.
+*/
+
+#import "Foundation/NSString.h"
+#import "Foundation/NSAttributedString.h"
+#import "Foundation/NSPersonNameComponents.h"
+#import "Foundation/NSPersonNameComponentsFormatter.h"
+
+@implementation NSPersonNameComponentsFormatter
+
+- (instancetype) init
+{
+  self = [super init];
+  if(self != nil)
+    {
+      _phonetic = NO;
+      _style = NSPersonNameComponentsFormatterStyleDefault;
+      _nameOptions = 0L;
+    }
+  return self;
+}
+
+- (void) _setNameOptions: (NSPersonNameComponentsFormatterOptions)opts
+{
+  _nameOptions = opts;
```

Comment:
I think the only option that is currently defined in the one for phonetic output. You could just set that value here and use it further down in stringFromPersonNameComponents: